### DISCUSSION TITLE
Add anchor snippet

### DISCUSSION THREAD

**Discussion by thejoyfuldev** [chakra-ui/chakra-ui#9270] [Ideas] [answered]
### Description

Request the addition of an Anchor snippet for a wrapper around framework specific links.

### Problem Statement/Justification

Chakra currently has a `LinkButton` component which has a comment to update with the framework's link component. However, this makes the component the size of a button and this is not always the intended desire. An `Anchor` component would wrap the current `Link` component around the framework's `Link` component.

### Proposed Solution or API

Proposed solution using `@tanstack/react-router`

```tsx
import { Link as BaseLink, LinkProps as BaseLinkProps } from '@chakra-ui/react'
import { Link, LinkProps } from '@tanstack/react-router'

export type AnchorProps = BaseLinkProps & LinkProps

export const Anchor = (props: AnchorProps) => {
  return (
    <BaseLink {...props} asChild>
      <Link href={props.href}>{props.children}</Link>
    </BaseLink>
  )
}
```

**NextJS**
```tsx
import { Link, LinkProps } from '@chakra-ui/react'
import NextLink, { LinkProps as NextLinkProps } from 'next/link'

export type AnchorProps = LinkProps & NextLinkProps

export const Anchor = (props: AnchorProps) => {
  return (
    <Link {...props} asChild>
      <NextLink passHref href={props.href}>
        {props.children}
      </NextLink>
    </Link>
  )
}
```

**Comment by segunadebayo** [comment, marked as answer]
You do not need an Anchor snippet. Wrap the framework link with `Link asChild`.

```tsx
<Link asChild>
  <NextLink href="/about">About</NextLink>
</Link>
```

That keeps `Link` styles without turning it into a `LinkButton`.
